free_fleet_adapter/tests/integration/test_nav1_tf_handler.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+
+from free_fleet_adapter.nav1_robot_adapter import Nav1TfHandler
+import rclpy
+from tf2_ros import Buffer
+
+import zenoh
+
+
+class TestNav1TfHandler(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+        cls.node = rclpy.create_node('test_nav1_tf_handler')
+        cls.zenoh_session = zenoh.open(zenoh.Config())
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.node.destroy_node()
+        cls.zenoh_session.close()
+        rclpy.shutdown()
+
+    def test_tf_does_not_exist(self):
+        tf_buffer = Buffer()
+
+        tf_handler = Nav1TfHandler(
+            'missing_nav1_tb3', self.zenoh_session, tf_buffer, self.node
+        )
+
+        transform_exists = False
+        for i in range(10):
+            transform = tf_handler.get_transform()
+            if transform is not None:
+                transform_exists = True
+                break
+            time.sleep(1)
```

Comment:
Is this integration test meant to be run with real hardware in the loop? If not, I'm surprised we'd have a sleep of one entire second in between pinging for transforms. I would think ~10ms should be enough wait if this is just testing against software.

Reply:
It was meant to give some time for initialization, as my initial tests in the past (without a zenoh router) too a while for peers to find each other and transforms to come through. Now that we encourage a zenoh router in the integration, it definitely does not warrant such a long sleep now. I've cleaned up all the tests a little bit, so it just waits for 2 seconds before checking. 1 second was not enough in my tests :sweat_smile: 

changes made in https://github.com/open-rmf/free_fleet/pull/145/commits/e3a5a1f10e6250ed34a1dfee816937f029e42d13